# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ContentIterator.h
# Commit: 5a92d3ac8a17
# Full Hash: 5a92d3ac8a176a1ad88ce23387f5143c0d09cd9c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - ContentSubtreeIterator #3: Fix a bug in ContentSubtreeIterator where the end container is in the light DOM of a shadow host r=jjaschke
#   
#   When the end container is in light DOM of a shadow host, we stop the
#   iteration after we have finished iterating the shadow root of it.
#   
# ==============================================================================

diff -r 058172014624 -r 5a92d3ac8a17 dom/base/ContentIterator.h
--- a/dom/base/ContentIterator.h	Wed May 07 15:05:09 2025 +0000
+++ b/dom/base/ContentIterator.h	Wed May 07 15:05:10 2025 +0000
@@ -121,6 +121,7 @@
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
           dom::AllowRangeCrossShadowBoundary::No,
+      nsTArray<AncestorInfo>* aInclusiveAncestorsOfEndContainer = nullptr);
   static nsIContent* GetPrevSibling(
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
